oak_functions/location_utils/src/lib.rs

```diff
@@ -14,239 +14,119 @@
-    pub position: Point,
+    let coverer = RegionCoverer {
+        min_level: level,
+        max_level: level,
+        level_mod: 0,
+        max_cells: 0,
+    };
+    let region = Cap::from_center_angle(&location.into(), radius);
+    Ok(coverer.covering(&region).0)
 }
 
-impl IndexValue {
-    pub fn from_bytes(c: &[u8]) -> IndexValue {
-        let mut value_key = [0; 8];
-        value_key.copy_from_slice(&c[0..8]);
-        IndexValue {
-            value_key,
-            position: Point::from_bytes(&c[8..16]),
-        }
+/// Finds the cell at the given `level` that covers the `location`.
+pub fn find_cell(location: &LatLng, level: u8) -> Result<CellID> {
+    let level = level as u64;
+    if level > s2::cellid::MAX_LEVEL {
+        anyhow::bail!("level too high");
     }
-
-    pub fn to_bytes(&self) -> Vec<u8> {
-        [self.value_key[..].to_vec(), self.position.to_bytes()].concat()
+    if level == s2::cellid::MAX_LEVEL {
+        return Ok(CellID::from(location));
     }
-}
 
-/// A relative location in cartesian coordinates.
-#[derive(Clone, Debug, Eq, PartialEq)]
-pub struct Point {
-    /// The relative x-coordinate in meters.
-    pub x: i32,
-    /// The relative y-coordinate in meters.
-    pub y: i32,
+    Ok(CellID::from(location).parent(level))
 }
 
-impl Point {
-    pub fn from_bytes(c: &[u8]) -> Point {
-        let mut x_bytes = [0; 4];
-        x_bytes.copy_from_slice(&c[0..4]);
-        let mut y_bytes = [0; 4];
-        y_bytes.copy_from_slice(&c[4..8]);
-
-        Point {
-            x: i32::from_be_bytes(x_bytes),
-            y: i32::from_be_bytes(y_bytes),
-        }
-    }
-
-    pub fn to_bytes(&self) -> Vec<u8> {
-        [self.x.to_be_bytes(), self.y.to_be_bytes()].concat()
-    }
-
-    /// Calculates the square of the distance between the point and another point.
-    ///
-    /// The square of the distance is sufficient for finding the minimum, seeing that `sqrt` grows
-    /// monotonically.
-    pub fn squared_distance(&self, other: &Point) -> i64 {
-        ((self.x - other.x) as i64).pow(2) + ((self.y - other.y) as i64).pow(2)
-    }
-
-    /// Validates that a point is within a cutoff radius.
-    pub fn validate_close_enough(&self, other: &Point, cutoff: i32) -> Result<(), String> {
-        let cutoff_sqaured = (cutoff as i64).pow(2);
-        if self.squared_distance(other) > cutoff_sqaured {
-            return Err(format!("closest data point is more than {}m away", cutoff));
-        }
-        Ok(())
-    }
+/// Converts a `LatLng` location to a byte representation.
+///
+/// The latitude and longitude for the location are converted microdegrees (`E6`) values. These
+/// values are then converted to big endian byte representations of the underlying 32 bit signed
+/// integer and concatenated.
+pub fn location_to_bytes(location: &LatLng) -> [u8; 8] {
+    let lat: E6 = location.lat.into();
+    let lng: E6 = location.lng.into();
```

Comment:
`oak_functions` input contains `lat` and `lon`
But here it's called `lng`.
Should we use the same abbreaviations everywhere?